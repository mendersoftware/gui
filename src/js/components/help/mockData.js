// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
