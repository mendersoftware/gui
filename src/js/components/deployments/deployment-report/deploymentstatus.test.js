import React from 'react';

import { defaultState, undefineds } from '../../../../../tests/mockData';
import { render } from '../../../../../tests/setupTests';
import DeploymentStatus, { DeploymentPhaseNotification } from './deploymentstatus';

describe('DeploymentStatus Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<DeploymentStatus deployment={defaultState.deployments.byId.d2} />);
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});

describe('DeploymentPhaseNotification Component', () => {
  const deployment = {
    ...defaultState.deployments.byId.d1,
    statistics: {
      ...defaultState.deployments.byId.d1.statistics,
      status: {
        ...defaultState.deployments.byId.d1.statistics.status,
        pause_before_committing: 0,
        pause_before_installing: 0,
        pause_before_rebooting: 1
      }
    }
  };
  it('renders correctly', async () => {
    const { baseElement } = render(<DeploymentPhaseNotification deployment={deployment} />);
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
