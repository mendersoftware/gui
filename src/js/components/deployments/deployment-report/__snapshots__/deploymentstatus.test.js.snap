// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentPhaseNotification Component renders correctly 1`] = `
.emotion-0 {
  background-color: #fdfdfd;
  padding: 16px 64px 16px;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.3929rem;
}

.emotion-1 iconButton {
  margin-right: 8px;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.1607rem;
  margin-left: 8px;
}

.emotion-2 iconButton {
  margin-right: 8px;
}

<div>
  <div
    class="emotion-0 flexbox center-aligned margin-bottom clickable "
    style="padding: 15px;"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
      data-testid="PauseIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19h4V5H6zm8-14v14h4V5z"
      />
    </svg>
    <div
      class="muted"
    >
      Deployment is 
      <span
        class="uppercased"
      >
        paused
      </span>
      . 
      <a>
        Review its status
      </a>
       to continue, retry or abort the deployment
       
    </div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall link-color emotion-2"
      data-testid="ArrowDropDownCircleOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 13-4-4h8z"
      />
    </svg>
  </div>
</div>
`;

exports[`DeploymentStatus Component renders correctly 1`] = `
.emotion-0 {
  background-color: #fdfdfd;
  padding: 16px 64px 16px;
}

.emotion-1 .key>b {
  background-color: #f7f7f7;
  color: rgba(0, 0, 0, 0.87);
}

<div>
  <div
    class="emotion-0 flexbox space-between centered margin-bottom "
  >
    <div
      class="flexbox column"
    >
      <div
        class="muted"
      >
        Status
      </div>
      <h4
        class="margin-bottom-none muted"
      >
        Pending
         (awaiting devices)
      </h4>
    </div>
    <div
      class="flexbox space-between align-right"
      style="min-width: 40%;"
    >
      <div
        class="flexbox column"
      >
        <div
          class="muted margin-bottom-small"
        >
          # devices
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="flexbox column"
      >
        <div
          class="muted margin-bottom-small"
        >
          Skipped
        </div>
        <div
          class="status"
        >
          0
        </div>
      </div>
      <div
        class="flexbox column"
      >
        <div
          class="muted margin-bottom-small"
        >
          Paused
        </div>
        <div
          class="status"
        >
          0
        </div>
      </div>
      <div
        class="flexbox column"
      >
        <div
          class="muted margin-bottom-small"
        >
          Pending
        </div>
        <div
          class="status"
        >
          1
        </div>
      </div>
      <div
        class="flexbox column"
      >
        <div
          class="muted margin-bottom-small"
        >
          In Progress
        </div>
        <div
          class="status"
        >
          0
        </div>
      </div>
      <div
        class="flexbox column"
      >
        <div
          class="muted margin-bottom-small"
        >
          Success
        </div>
        <div
          class="status"
        >
          0
        </div>
      </div>
      <div
        class="flexbox column"
      >
        <div
          class="muted margin-bottom-small"
        >
          Fail
        </div>
        <div
          class="status"
        >
          0
        </div>
      </div>
    </div>
    <div
      class="break-all two-columns emotion-1  column-data "
      style="align-items: center; align-self: flex-start; grid-template-columns: max-content 1fr; max-width: 25vw;"
    >
      <div
        class="align-right  muted"
      >
        <b>
          Max attempts per device
        </b>
      </div>
      <div
        class="flexbox "
      >
        <div
          title="1"
        >
          1
        </div>
      </div>
      <div
        class="align-right  muted"
      >
        <b>
          Maximum number of devices
        </b>
      </div>
      <div
        class="flexbox "
      >
        <div
          title="N/A"
        >
          N/A
        </div>
      </div>
    </div>
  </div>
</div>
`;
